app/controllers/register/establishers/company/CompanyRegistrationNumberUpdateController.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.register.establishers.company
+
+import config.FrontendAppConfig
+import controllers.actions._
+import controllers.register.CompanyRegistrationNumberBaseController
+import forms.CompanyRegistrationNumberFormProvider
+import identifiers.register.establishers.company.CompanyRegistrationNumberId
+import javax.inject.Inject
+import models.requests.DataRequest
+import models.{CompanyRegistrationNumber, Index, Mode}
+import play.api.data.Form
+import play.api.i18n.MessagesApi
+import play.api.mvc.{AnyContent, Call}
+import services.UserAnswersService
+import utils._
+import utils.annotations.EstablishersCompany
+import views.html.register.companyRegistrationNumberUpdate
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class CompanyRegistrationNumberUpdateController @Inject()(
+                                                     val appConfig: FrontendAppConfig,
+                                                     override val messagesApi: MessagesApi,
+                                                     userAnswersService: UserAnswersService,
+                                                     @EstablishersCompany navigator: Navigator,
+                                                     authenticate: AuthAction,
+                                                     getData: DataRetrievalAction,
+                                                     allowAccess: AllowAccessActionProvider,
+                                                     requireData: DataRequiredAction,
+                                                     formProvider: CompanyRegistrationNumberFormProvider
```

Comment:
Will this work for variations, where there will be no boolean, i.e. hasCrn?

Reply:
I want to test this in the other related story